src/Kestrel.Core/Internal/Http/HttpProtocol.FeatureCollection.cs

```diff
@@ -90,8 +90,21 @@ string IHttpRequestFeature.Scheme
 
         string IHttpRequestFeature.Method
         {
-            get => Method;
-            set => Method = value;
+            get
+            {
+                if (_customMethod != null)
+                {
+                    return _customMethod;
+                }
+
+                _customMethod = HttpUtilities.MethodToString(Method);
```

Comment:
Why store this in _costomMethod? It's a static lookup. You store null for Set.

Reply:
`GetKnownMethod` returns an `HttpMethod` enum for known methods, but not the string so null.

The field is there anyway for a non-known method, so once you've looked it up once, might as well store it if its checked again rather than doing another lookup,

Reply:
So why didn't you store it in the Set code path?

Reply:
If you're going to use it for both then call it `_methodText`

Reply:
Ahh.. that's a very good point 😄